src/write/line.rs

```diff
@@ -1003,177 +1018,378 @@ mod convert {
+                Some(file) => Some(from_dwarf.attr_line_string(file.path_name())?),
+                None => None,
+            };
+            let mut convert = LineConvert::new(
+                from_dwarf,
+                from_program,
+                comp_name,
+                encoding,
+                line_encoding,
+                line_strings,
+                strings,
+            )?;
+
+            while let Some(row) = convert.read_row()? {
+                match row {
+                    LineConvertRow::SetAddress(address, row) => {
+                        if convert.in_sequence() {
+                            return Err(ConvertError::InvalidAddress);
+                        }
+                        let address =
+                            convert_address(address).ok_or(ConvertError::InvalidAddress)?;
+                        convert.begin_sequence(Some(address));
+                        convert.generate_row(row);
+                    }
+                    LineConvertRow::Row(row) => {
+                        if !convert.in_sequence() {
+                            convert.begin_sequence(None);
+                        }
+                        convert.generate_row(row);
+                    }
+                    LineConvertRow::EndSequence(length) => {
+                        if !convert.in_sequence() {
+                            convert.begin_sequence(None);
+                        }
+                        convert.end_sequence(length);
+                    }
+                }
+            }
+            Ok(convert.program())
+        }
+    }
+
+    /// The result of [`LineConvert::read_row`].
+    #[derive(Debug)]
+    pub enum LineConvertRow {
+        /// A row that used a `DW_LNS_set_address` instruction.
+        ///
+        /// This is expected to be the first row in a sequence,
+        /// but [`LineConvert::read_row`] doesn't enforce that.
+        ///
+        /// This row will have its `address_offset` field set to 0.
+        /// All subsequent rows in the sequence will have their `address_offset`
+        /// field set to an offset from this address.
+        SetAddress(u64, LineRow),
+        /// A row produced by the line number program.
+        Row(LineRow),
+        /// The address offset of the end of the sequence.
+        EndSequence(u64),
+    }
+
+    /// The result of [`LineConvert::read_sequence`].
+    #[derive(Debug)]
+    pub struct LineConvertSequence {
+        /// The address of the first instruction in the sequence.
+        pub start: Option<u64>,
+        /// The offset in bytes of the next instruction after the sequence.
+        pub length: u64,
+        /// The rows in the sequence.
+        ///
+        /// The `LineRow::address` fields are set to an offset from `Self::start`.
+        pub rows: Vec<LineRow>,
+    }
+
+    /// The state for the conversion of a line number program.
+    #[derive(Debug)]
+    pub struct LineConvert<'a, R: Reader> {
```

Comment:
Is this intended for use outside of `LineProgram::from`? Because from a user-perspective, these docs don't tell me much about why I'd use this or how.